wallet/src/main/scala/org/bitcoins/wallet/Wallet.scala

```diff
@@ -531,14 +523,15 @@ abstract class Wallet
     * @return
     */
   override def createNewAccount(kmParams: KeyManagerParams): Future[Wallet] = {
-    val lastAccountOptF = accountDAO
-      .findAll()
-      .map(_.filter(_.hdAccount.purpose == kmParams.purpose))
-      .map(_.sortBy(_.hdAccount.index))
-      // we want to the most recently created account,
-      // to know what the index of our new account
-      // should be.
-      .map(_.lastOption)
+    def lastAccountOptF =
```

Comment:
make this another private method at the top level of the class rather than embedding inside of `createNewAccount`. Maybe it's useful elsewhere in the future.

Reply:
👍